lib/common/xml.c

```diff
@@ -1147,7 +1071,7 @@ pcmk__xml_needs_escape(const char *text, enum pcmk__xml_escape_type type)
                 break;
         }
 
-        text++;
+        text = g_utf8_next_char(text);
```

Comment:
We could alternatively use `xmlGetUTF8Size()`. I've considered that, since the purpose of all this is to escape text that will be part of an XML tree, and one could argue that we should use libxml2's logic to the extent possible.

I used `g_utf8_next_char()` for a few reasons:
1. I found it first
2. It performs better because it doesn't validate the encoding
3. It doesn't require a temporary "length" output variable
4. In general (knock on wood) GLib seems to be less buggy than libxml2.

In an upcoming PR I use `xmlGetUTF8Char()` (not to be confused with `xmlGetUTF8Size()`) in an updated version of `crm_xml_sanitize_id()`.